services/wallet/transfer/commands.go

```diff
@@ -355,10 +359,36 @@ func (c *transfersCommand) Run(ctx context.Context) (err error) {
 		return err
 	}
 
-	err = c.db.SaveTranfers(c.chainClient.ChainID, c.address, allTransfers, []*big.Int{c.block})
-	if err != nil {
-		log.Error("SaveTranfers error", "error", err)
-		return err
+	// Update MultiTransactionID from pending entry
+	for index := range allTransfers {
+		transfer := &allTransfers[index]
+		if transfer.MultiTransactionID == NoMultiTransactionID {
+			entry, err := c.transactionManager.GetPendingEntry(c.chainClient.ChainID, transfer.ID)
+			if err != nil {
+				if err == sql.ErrNoRows {
+					log.Warn("Pending transaction not found for", c.chainClient.ChainID, transfer.ID)
+				} else {
+					return err
+				}
+			} else {
+				transfer.MultiTransactionID = entry.MultiTransactionID
+				if transfer.Receipt != nil && transfer.Receipt.Status == types.ReceiptStatusSuccessful {
+					// TODO: Nim logic was deleting pending previously, should we notify UI about it?
+					err := c.transactionManager.DeletePending(c.chainClient.ChainID, transfer.ID)
```

Comment:
Did u check with mobile team that this won't break anything on their side?

Reply:
@rasom please have a look at how this will line change impact on the mobile side. You should have a `DeletePending` statusgo RPC call on your side, which should be redundant and fail after this change

Reply:
We haven't started re-implementation of the wallet UI at the moment, so as long as you don't change signatures in `mobile/status.go` (`status-go`'s non RPC API stays backward compatible) it is unlikely it will have any impact on mobile side at the moment.